tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/hci.h>
+#include "mesh_test.h"
+#include "mesh/adv.h"
+
+#define LOG_MODULE_NAME test_adv
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_INF);
+
+#define WAIT_TIME 60 /*seconds*/
+
+struct bt_mesh_test_adv {
+	uint8_t retr;
+	int64_t interval;
+};
+
+static struct bt_mesh_send_cb send_cb;
+static struct bt_mesh_test_adv *xmit_param;
+static struct k_sem observer_sem;
+static const char *txt_msg = "adv test";
+static const char *cb_msg = "cb test";
+static int64_t tx_timestamp;
+static int seq_checker;
+
+static void test_tx_init(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+}
+
+static void test_rx_init(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+}
```

Comment:
Three remarks:
- Neither of these seems to be used?
- If they are meant to be used, could it not be one function since they are identical?
- Is it neccesary with oneliner functions here?

Reply:
It is provided as pre initialization here: https://github.com/zephyrproject-rtos/zephyr/pull/50050/files#diff-27be331c9da5ebf00b50fdde2d30791a6fd470e79c5f65fa3fde138dc506a8a5R304

The functions provide time into `bst_ticker_set_next_tick_absolute` of bsim engine to interpret fail behavior. Additionally, it provides configuration for mesh devices. Since the test suite is based on advertiser\scanner and doesn't require mesh configuration I provide `NULL`. Each virtual bsim device requires its own pre init function.